inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp

```diff
@@ -942,6 +942,64 @@ void convertFunctionToICNNNetwork(const std::shared_ptr<const ::ngraph::Function
     const ngraph::NodeVector& nodes = graph->get_ops();
     bool keep_constants = keep_constant_inputs || ::ngraph::op::util::has_op_with_type<::ngraph::op::FakeQuantize>(graph);
 
+    std::unordered_map<std::string, std::shared_ptr<ngraph::Node>> unique_names;
+    auto can_change_name = [](const std::shared_ptr<ngraph::Node> & node) -> bool {
+        if (ngraph::as_type_ptr<ngraph::op::Parameter>(node) ||
+            ngraph::as_type_ptr<ngraph::op::Result>(node)) {
+            return false;
+        }
+        for (const auto & output : node->outputs()) {
+            for (const auto & consumer : output.get_target_inputs()) {
+                if (ngraph::is_type<ngraph::op::Result>(consumer.get_node())) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    };
+
+    auto generate_unique_name = [&unique_names](std::string name) -> std::string {
+        int64_t suffix = 1;
+        while(unique_names.count(name + "/" + std::to_string(suffix))) {
+            ++suffix;
+        }
+        return name + "/" + std::to_string(suffix);
```

Comment:
In 99% cases original models doesn't have nodes with duplicates names so if we generated non-unique name during transformations we will do it only once per name. So it will require a single iteration in this names generator. And would prefer to see small numbers in final net rather than have a giant numbers.